File: Haskell/ABC/ABC101/ABC101_B_v01.hs
-- [contest]  ABC 101
-- [task]     B
-- [URL]      https://atcoder.jp/contests/abc101/tasks/abc101_b
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc101/submissions/6099337 ; AC

-- [reference]
-- http://hackage.haskell.org/package/bytestring-0.10.8.2/docs/Data-ByteString-Char8.html
-- https://qiita.com/hnw/items/3f7d27b742c5a1a99a9a
-- https://myuon.github.io/posts/haskell-atcoder/

-- modules to import --

import Data.ByteString.Char8
import Data.Maybe
import Prelude

-- functions for this task --

read_Integer = fst . fromJust . Data.ByteString.Char8.readInteger
get_Integer  = read_Integer <$> Data.ByteString.Char8.getLine

calculate_digit_sum :: Integer -> Integer
calculate_digit_sum target
        | target >= 10 = (mod target 10) + (calculate_digit_sum $ div target 10)
        | target >=  0 = target  

task_B :: Integer -> Bool
task_B target
        | mod target (calculate_digit_sum target) == 0 = True
        | otherwise                                    = False

-- main process is below --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given integer
        integer_given <- get_Integer

        -- STEP.02
        -- output the answer of this task
        Prelude.putStrLn $ if (task_B $ integer_given) then "Yes" else "No"

-- End of this source code --

File: C(GCC5.4.1)/ABC/ABC101/ABC101_B_v01.c
// [contest]  ABC 101
// [task]     B
// [URL]      https://atcoder.jp/contests/abc101/tasks/abc101_b
// [compiler] C (GCC 5.4.1)
// [status]   https://atcoder.jp/contests/abc101/submissions/5831444 : CE
//            https://atcoder.jp/contests/abc101/submissions/5831451 : AC

#include <stdio.h>
#include <stdbool.h>

int calculate_digit_sum (int target)
{
    // return value of this function
    int digit_sum;

    // STEP.01
    // initialize the variable to store the digit sum
    digit_sum = (int)0;

    // STEP.02
    // calculate the digit sum
    while (target > (int)0)
    {
        digit_sum += target % (int)10;
        target    /= (int)10;
    }

    // STEP.END
    // declare the return value of this function
    return digit_sum;
}

bool is_multiple (int target, int base)
{
    if (target % base == (int)0) { return true;  }
    else                         { return false; }
}

int main (void)
{
    // variables for `main`
    int int_given;
    int rtvl_scanf;

    // STEP.01
    // read out the given string
    rtvl_scanf = scanf("%d", &int_given);

    // STEP.02
    // output whether the digit sum of given integer is the multiple of the given integer or not
    if ( is_multiple( int_given, calculate_digit_sum(int_given) ) )
    { printf("%s\n", "Yes"); }
    else
    { printf("%s\n", "No"); }

    // STEP.END
    return 0;
}